### PR TITLE
OCPBUGS-37598: V1.36.1 rebase master

### PR DESCRIPTION
**Pull request by arkadeepsen**
Rebase to upstream v1.36.1

- Ran the following script to create the rebase branch:

  ```
  ./UPSTREAM-MERGE.sh v1.36.1
  ```
- Regenerated the `patches/03-setversion.patch` file to reflect the correct `SIMPLE_VERSION`
- Regenerated the fixsanity patch
- Regenerated the disable security context patch
- Regenerated the `06-build-darwin-binary.patch` file
- Regenerate the `11-openshiftv1-cli.patch` file
- Removed the downstream ansible code
- Updated vendor directory by running the following commands:
  ```
  go mod tidy
  go mod vendor
  ```

### CODE REVIEW

**Comment by arkadeepsen** [comment]
/hold
until https://issues.redhat.com/browse/CFE-1098 is completed

**Comment by KeenonLee** [comment]
```
jitli@RedHat:~/work/src/github/sdk/downstream/ocp-release-operator-sdk/build$ ./operator-sdk version
operator-sdk version: "v1.35.0-ocp", commit: "9cbe6d88856086ff9df288da77b4b6dcf656f331", kubernetes version: "v1.29.0", go version: "go1.22.0", GOOS: "darwin", GOARCH: "arm64"
```
Still  "v1.35.0-ocp"

**Comment by arkadeepsen** [comment]
> ```
> jitli@RedHat:~/work/src/github/sdk/downstream/ocp-release-operator-sdk/build$ ./operator-sdk version
> operator-sdk version: "v1.35.0-ocp", commit: "9cbe6d88856086ff9df288da77b4b6dcf656f331", kubernetes version: "v1.29.0", go version: "go1.22.0", GOOS: "darwin", GOARCH: "arm64"
> ```
> 
> Still "v1.35.0-ocp"

Updated the `patches/03-setversion.patch` file to reflect the correct version `v1.36.0-ocp`

File: patches/13-kubernetes-core-collection-bump.patch [hunk 1]
```diff
@@ -1,379 +0,0 @@
...
-diff --git a/internal/helm/release/manager.go b/internal/helm/release/manager.go
-index 07641ccc7..72a246461 100644
---- internal/helm/release/manager.go
-+++ internal/helm/release/manager.go
-@@ -95,7 +95,7 @@ func (m manager) IsUpgradeRequired() bool {
- 
- // Sync ensures the Helm storage backend is in sync with the status of the
- // custom resource.
--func (m *manager) Sync(ctx context.Context) error {
-+func (m *manager) Sync(_ context.Context) error {
```
Comment (TrilokGeer) [inline patches/13-kubernetes-core-collection-bump.patch:175]: There are non-ansible changes in the patch. Are these still required and added as part of another patch file?

Comment (arkadeepsen) [reply patches/13-kubernetes-core-collection-bump.patch:175]: This patch file is not required as most of the changes in the patch are already added in the upstream code. Secondly, these patches were also not related to the kubernetes.core bump. As we have moved the code related to ansible-operator to ansible-operator-plugins, these patch file is no longer required. I have removed the file in  https://github.com/openshift/ocp-release-operator-sdk/pull/387/commits/6aae506ab65bb7aab8e07662166ec9685367cd6d.

File: ci/tests/e2e-helm.sh [hunk 2]
```diff
@@ -29,6 +29,7 @@ oc version
 echo $ROOTDIR
 
 # install operator-sdk
+export GOBIN=${GOPATH}/bin
```
Comment (arkadeepsen) [inline ci/tests/e2e-helm.sh:32]: Before `make install` there's an additional check if the `GOBIN` is set or not. For that reason this export is needed.
https://github.com/openshift/ocp-release-operator-sdk/pull/387/commits/011460111ef73896e64d17e4329c6b0407835ba1#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R76-R79

Comment (TrilokGeer) [reply ci/tests/e2e-helm.sh:32]: A conditional check is required to set GOBIN only if there is no other value inherited from env.

Comment (arkadeepsen) [reply ci/tests/e2e-helm.sh]: Added the check before setting the value of GOBIN.

File: patches/06-build-darwin-binary.patch [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-@@ -195,3 +195,9 @@ help: ## Show this help screen.
+--- ./Makefile.darwin	2024-07-25 17:30:44.164036603 +0530
++++ ./Makefile	2024-07-25 17:32:13.983235216 +0530
+@@ -208,3 +208,9 @@ help: ## Show this help screen.
  	@echo 'Available targets are:'
  	@echo ''
  	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-25s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 +
 +.PHONY: build-darwin
-+build-darwin: ## Build operator-sdk, ansible-operator, and helm-operator.
```
Comment (chiragkyal) [inline patches/06-build-darwin-binary.patch:10]: Are we removing `helm-operator` because the command itself is not calling it?

Comment (arkadeepsen) [reply patches/06-build-darwin-binary.patch:10]: The bug (https://bugzilla.redhat.com/show_bug.cgi?id=1930356) was originally reported for only operator-sdk binary. As you can see, the only operator-sdk is being built by the make target. I think that when the patch was created originally (https://github.com/openshift/ocp-release-operator-sdk/pull/104), the comment was simply copied from the build target in the main Makefile, however, the target is only used for building operator-sdk.

**Comment by chiragkyal** [comment]
Should we update the commit messages with `UPSTREAM: <carry>:` prefix ?

**Comment by arkadeepsen** [comment]
> Should we update the commit messages with `UPSTREAM: <carry>:` prefix ?

There are no previous commits with such prefix in this repo. I am using the same convention in this PR. However, I am open to updating the commit messages with the prefix. @TrilokGeer WDYT?

**Comment by TrilokGeer** [comment]
> > Should we update the commit messages with `UPSTREAM: <carry>:` prefix ?
> 
> There are no previous commits with such prefix in this repo. I am using the same convention in this PR. However, I am open to updating the commit messages with the prefix. @TrilokGeer WDYT?

https://github.com/openshift/enhancements/blob/master/dev-guide/kubernetes-rebase.md#picking-commits-from-the-previous-rebase-branch-to-the-new-branch

**Comment by arkadeepsen** [comment]
> > > Should we update the commit messages with `UPSTREAM: <carry>:` prefix ?
> > 
> > 
> > There are no previous commits with such prefix in this repo. I am using the same convention in this PR. However, I am open to updating the commit messages with the prefix. @TrilokGeer WDYT?
> 
> https://github.com/openshift/enhancements/blob/master/dev-guide/kubernetes-rebase.md#picking-commits-from-the-previous-rebase-branch-to-the-new-branch

Updated the commit messages

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved

File: internal/plugins/openshift/v1/init.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ import (
 
 const (
 	// The current OCP release version.
-	ocpProductVersion = "4.15"
+	ocpProductVersion = "4.18"
 	// The currently used version of ubi8/ubi-minimal images.
 	ubiMinimalVersion = "8.8"
```
Comment (TrilokGeer) [inline internal/plugins/openshift/v1/init.go:32]: Is there any update required for ubi version?

Comment (TrilokGeer) [reply internal/plugins/openshift/v1/init.go:32]: The image version has known security vulnerabilites.

Comment (arkadeepsen) [reply internal/plugins/openshift/v1/init.go:32]: The latest ubi8 tag available in `registry.access.redhat.com/ubi8/ubi-minimal` and `registry.access.redhat.com/ubi8/ubi-micro` is 8.10. Should I update to 8.10?

Comment (arkadeepsen) [reply internal/plugins/openshift/v1/init.go]: Updated to 8.10.

**Comment by TrilokGeer** [comment]
The latest tags on 8.10 are not reporting any important vulnerabilities, approving the PR based on present day CVE scan reports.

/lgtm

**Pull request merged**